### DISCUSSION TITLE
Unable to recognize the installed custom module?

### DISCUSSION THREAD

**Discussion by happy-berry** [processone/ejabberd#4155] [Questions] [answered]
OS:
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:        12
Codename:       bookworm

ejabberd version : 23.10

install step:  wget https://github.com/processone/ejabberd/releases/download/23.10/ejabberd-23.10-1-linux-x64.run

I have followed the official documentation for the operation ， but it seems unable to find the module..

code : 
```erlang
-module(mod_hello_world).

-behaviour(gen_mod).

%% Required by ?INFO_MSG macros
-include("logger.hrl").

%% Required by ?T macro
-include("translate.hrl").

%% gen_mod API callbacks
-export([start/2, stop/1, depends/2, mod_options/1, mod_doc/0]).

start(_Host, _Opts) ->
    ?INFO_MSG("Hello, ejabberd world!", []),
    ok.

stop(_Host) ->
    ?INFO_MSG("Bye bye, ejabberd world!", []),
    ok.

depends(_Host, _Opts) ->
    [].

mod_options(_Host) ->
    [].

mod_doc() ->
    #{desc =>
          ?T("This is an example module.")}.
```
my step : 
1、mkdir -p $HOME/.ejabberd-modules/sources/mod_hello_world/src/
2、cp my_er/mod_hello_world.erl $HOME/.ejabberd-modules/sources/mod_hello_world/src/
3、echo "summary: 'Hello World example module'" > $HOME/.ejabberd-modules/sources/mod_hello_world/mod_hello_world.spec
Then I execute ejabberdctl modules_available . Why can't I get anything available?

**Comment by licaon-kter** [comment]
you've read https://github.com/processone/ejabberd-contrib/#ejabberd-contrib ?

**Reply by happy-berry** [reply]
No.  I found it on this link  https://docs.ejabberd.im/developer/extending-ejabberd/modules/ ---》 add your module.  It just met what I needed....but when I configure it, everything is not as I imagined
（I hope to write a module that can help me intercept messages sent to my HTTP server, and then set the results I return to Packet）

**Reply by happy-berry** [reply]
I am a beginner， How should I configure it? Can you help me

**Comment by badlop** [comment, marked as answer]
Is ejabberd running under your user account? Maybe `$HOME` of the ejabberd program is not your personal `$HOME`.

Does ejabberd store and read modules from that path? Or maybe ejabberd is using a different path?

One method to know the path that ejabberd is using for those modules:

```
ejabberdctl debug
ext_mod:modules_dir().

"/home/badlop/.ejabberd-modules"
```

Another method is to run `ejabberdctl modules_update_specs`, and search in the `$HOME` of the user that runs ejabberd the directory .`ejabberd-modules`.
